src/ZkFold/Base/Algebra/Basic/Class.hs

```diff
@@ -220,26 +475,36 @@ instance BinaryExpansion Bool where
     fromBinary [x] = x
     fromBinary _   = error "fromBits: This should never happen."
 
---------------------------------------------------------------------------------
-
-instance AdditiveSemigroup a => AdditiveSemigroup [a] where
-    (+) = zipWith (+)
+instance MultiplicativeMonoid a => Exponent Bool a where
+    _ ^ False = one
+    x ^ True  = x
 
-instance AdditiveMonoid a => AdditiveMonoid [a] where
-    zero = repeat zero
-
-instance AdditiveGroup a => AdditiveGroup [a] where
-    negate = map negate
+--------------------------------------------------------------------------------
 
 instance MultiplicativeSemigroup a => MultiplicativeSemigroup [a] where
     (*) = zipWith (*)
 
+instance Exponent b a => Exponent b [a] where
+    x ^ p = map (^ p) x
+
 instance MultiplicativeMonoid a => MultiplicativeMonoid [a] where
     one = repeat one
 
 instance MultiplicativeGroup a => MultiplicativeGroup [a] where
     invert = map invert
 
+instance AdditiveSemigroup a => AdditiveSemigroup [a] where
+    (+) = zipWith (+)
+
+instance Scale b a => Scale b [a] where
+    scale = map . scale
```

Comment:
How about generalising it to Functor? This way polynomials which have Vectors under the hood will get Scale almost for free.
```suggestion
instance Functor f => Scale b a => Scale b (f a) where
    scale = fmap . scale
```

Reply:
This is a great idea, however, once we make an `ArithmeticCircuit` into a functor, this instance would do the wrong thing for it.
(But maybe this is an argument against making `ArithmeticCircuit` a functor, @vlasin, @iokasimov, what do you think?)

Reply:
Yes, for now, it is not clear if `ArithmeticCicruit` should be a functor. I like the proposed generalization.

Reply:
@vlasin @TurtlePU Probably it's possible to make it a `Functor` instance if we make `acWitness` a continuation.

Reply:
@iokasimov, it is, but then a `Scale` instance for functors would silently do the wrong thing for it if one doesn't import the correct import from Instances module